CMakeLists.txt

```diff
@@ -30,6 +31,34 @@ endif()
 
 cmake_minimum_required(VERSION 3.16)
 
+# search for Vulkan SDK
+find_package(Vulkan)
+
+if(VULKAN_FOUND)
+    if(NOT Vulkan_dxc_EXECUTABLE)
+        if(WIN32)
+            find_program(Vulkan_dxc_EXECUTABLE
+                        NAMES dxc
+                        HINTS
+                        "$ENV{VULKAN_SDK}/Bin"
+                        )
+        else()
+            find_program(Vulkan_dxc_EXECUTABLE
+                        NAMES dxc
+                        HINTS
+                        "$ENV{VULKAN_SDK}/bin"
+                        )
+        endif()
+    endif()
+    if(DEFINED Vulkan_dxc_EXECUTABLE)
+        message("Found DirectX Shader Compiler under ${Vulkan_dxc_EXECUTABLE}")
+    elseif()
+        message("Couldn't find DirectX Shader Compiler executable, make sure it is present in Vulkan SDK or add it manually via Vulkan_dxc_EXECUTABLE env variable")
+    endif()
+else()
+    message("VULKAN_SDK path not found, HLSL shaders won't be compiled")
```

Comment:
Is it possible to manually specify a DXC path, or does it always fail if no SDK is installed? I often use a different DXC version that's newer than the SDK, so having a way to optionally manually specify the path would be great.

Reply:
My reading is that you could specify `Vulkan_dxc_EXECUTABLE` on the cmake configure command line and it would be honored.  

Also, `find_program` does in fact check the system path, unless `NO_DEFAULT_PATH` or `NO_SYSTEM_ENVIRONMENT_PATH` is passed to `find_program` (or `CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH` is set to false).

Reply:
Also form cmake version 3.25, `Vulkan_dxc_EXECUTABLE` is also added by `FindVulkan`.
https://cmake.org/cmake/help/latest/module/FindVulkan.html